---
title: "Research"
permalink: /research/
---

## Research Interests
- Computer Science: Large language models, Machine learning, Video LLMs, SLMs, Artificial intelligence, Computer vision
- Software Engineering: Data strucures and algorithms, Object-oriented programming, Design patterns, Distributed systems, Databases, and Cloud computing
- Mathematics: Topology, Game theory, Linear Algebra, and Discrete mathematics.
- Inter-disciplinary: Robotics, Human-Computer Interface, AI in drug discovery, Computation social science, Computational Biology, Binformatics, Personalized medicines, and Algorithmic Game theory.

## Current Projects
- **Reducing hallucination in text to workflow generation using Large Language Models**: A short description or abstract.  
  - Abstract submitted. It is a work in progress.
- **Text to terraform templates generation**: A short description or abstract.  
  - Abstract submitted. It is a work in progress.
 
## Past Projects
- **SNP Detection in Liquid Biopsy**: Machine learning models for cancer prognosis.  
  - Variation when compared to a reference genome (i.e. hg19) can happen at
various different frequencies. For instance, in somatic cells, variations occur
less frequently (that is, one in thousand reads supports the variation).
The errors made by the Alignment process, PCR process and the Sequencing
process could can introduce variations of the same frequency. These
errors can be construed as noise. This noise could be mislabelled as cancer,
and give rise to erroneous SNP calls. Hence, the prime objective is to
develop a machine learning model to differentiate a variation in a read due
to cancer from a variation due to noise (generated mainly due to errors in
sequencing and/or alignment and/or PCR process) in a query. Cancer analysis
using genomic sequence requires accurate identification of somatic
and germline variants in sequencing data. Manual intervention to refine
the variant calls is required for applying the SNP filters. However, manual
variant SNP filter is tedious, expensive in terms of manual efforts, very unstructured,
not easily replicable, and highly reliant on read features solely.
Hence, we use ourMLmodel (i.e. Random Forest) to automate the process,
and consider base specific features to increase accuracy. The final model
takes in reads at a pileup columns (i.e. given base position) and assigns a
probability score for each base to filter out erroneous bases. And this probability
score is used to more accurate SNP calls. Thus, this model labels test
data set with high specificity and sensitivity. The model improves on the
manual SNP calling and/or aids the same.



I actively seek to collaborate with others in these areas. Please feel free to reach out if you're interested in discussing research or potential collaborations.
